test/make_dist.py

```diff
@@ -156,16 +157,23 @@ def download_dist():
     return tar_path
 
 
-def make_dist():
+def make_dist(download_only=False):
     # first try to download a pre-generated dist tarball; this is a lot faster
     # but these tarballs are built for production NPM mode
     source = None
     if os.getenv("NODE_ENV") != "development":
         source = download_dist()
     if not source:
-        source = build_dist()
+        if not download_only:
+            source = build_dist()
+        else:
+            print("make_dist: Download failed: artifact does not exist")
+            sys.exit(1)
     return source
 
 
 if __name__ == '__main__':
-    print(make_dist())
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--download-only', action='store_true', help="Download or build release tarball")
```

Comment:
This should be the help for the particular -d option, thus "Fail instead of build locally if download is not available" or something such.

Reply:
Also, please also add back `-d`. Short options are more convenient to type. I just like to use long options in widespread code, as they are easier to understand when reading.